CRM/Contribute/Form/ManagePremiums.php

```diff
@@ -291,10 +292,16 @@ public function postProcess() {
     $this->_processImages($params);
 
     // Save the premium product to database
-    $premium = CRM_Contribute_BAO_Product::create($params);
+    // If we're updating, we need to pass in the premium product Id
+    if ($this->_action & CRM_Core_Action::UPDATE) {
+      $premium = Product::update(FALSE)->setValues($params)->addWhere('id', '=', $this->_id)->execute()->first();
+    }
+    else {
+      $premium = Product::create(FALSE)->setValues($params)->execute()->first();
+    }
```

Comment:
Or you could just turn these 6 lines into a 1-liner by using 
```suggestion
    $premium = Product::save(FALSE)->addRecord($params)->execute()->first();
```